include/nonkeyedcommand.h

```diff
@@ -0,0 +1,144 @@
+/*
+ * BSD 2-Clause License
+ *
+ * Copyright (c) 2021, Hewlett Packard Enterprise
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SMARTREDIS_NONKEYEDCOMMAND_H
+#define SMARTREDIS_NONKEYEDCOMMAND_H
+
+#include "command.h"
+
+using parsed_reply_map = std::unordered_map<std::string, std::string>;
+using parsed_reply_nested_map = std::unordered_map<std::string, parsed_reply_map>;
+
+///@file
+
+namespace SmartRedis {
+
+class RedisServer;
+
+/*!
+*   \brief The NonKeyedCommand intermediary class constructs Client
+*          commands without keys. These commands use db node addresses.
+*   \details The KeyedCommand class has multiple methods for dealing
+*            with non-keyed commands.
+*/
+class NonKeyedCommand : public Command
+{
+    public:
+        /*!
```

Comment:
Same comment as the ``KeyedCommand`` classes.  We should explicitly declare the constructor, destructor, move assignment, copy assignment, move constructor, and copy constructor for each class.